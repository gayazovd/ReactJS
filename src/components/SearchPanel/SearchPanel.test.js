import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SearchPanelView from './view';

configure({ adapter: new Adapter() });

describe('component <SearchPanelView /> is render', () => {
    const onInputChange = jest.fn(),
        onClick = jest.fn(),
        props = {
            search: '',
            onInputChange,
            onClick,
            tabs: [1, 2],
            activeTab: 1
        };
    const searchPanel = shallow(<SearchPanelView {...props} />);
    it('get snapshot <SearchPanelView /> component', () => {
        expect(searchPanel).toMatchSnapshot();
    });

    it('onClick called', () => {
        const output = searchPanel.find('.search');
        output.simulate('click');
        expect(onClick).toHaveBeenCalled();
    });

    it('onInputChange called and change', () => {
        const output = searchPanel.find('.inputButton');
        output.simulate('change', {
            target: { search: 'aaaa' }
        });
        expect(onInputChange).toHaveBeenCalledWith({ target: { search: 'aaaa' } });
    });
});
