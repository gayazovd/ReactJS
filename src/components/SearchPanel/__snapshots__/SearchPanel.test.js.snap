// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <SearchPanelView /> is render get snapshot <SearchPanelView /> component 1`] = `
<div
  className="wrapper"
>
  <div
    className="logo"
  >
    netflixroullete
  </div>
  <div
    className="titleFind"
  >
    FIND YOUR MOVIE
  </div>
  <input
    className="inputButton"
    onChange={[MockFunction]}
    value=""
  />
  <div
    className="buttonsWraper"
  >
    <div
      className="optionsButtons"
    >
      <p
        className="title"
      >
        SEARCH BY
      </p>
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
      />
      <button
        className="buttons"
        onClick={[Function]}
        type="button"
      />
    </div>
    <button
      className="search"
      onClick={[Function]}
      type="button"
    >
      SEARCH
    </button>
  </div>
</div>
`;
